parla/task_runtime.py

```diff
@@ -1010,17 +1010,15 @@ def run(self) -> None:
                         logger.debug(
                             f"[WorkerThread %d] Starting: %s", self.index, self.task.name)
                         self._status = "Running Task {}".format(self.task)
+                        self.scheduler.incr_running_tasks() 
                         self.task.run()
+                        self.scheduler.decr_running_tasks()
 
                         # Free self back to worker pool
                         self._remove_task()
                         self.scheduler.append_free_thread(self)
```

Comment:
I think freeing the thread before calling the callbacks is correct. 
I know we talked about it earlier, but I'm pretty sure this is good as then itself is available to be assigned in the launching phase.

Reply:
I see. I think I mispoint to this part for other errors. So the current state is correct, isn't it?